// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.confidential_match.lookup_server.proto_backend;

option java_package = "com.google.cm.lookupserver.backend";
option java_outer_classname = "LocationProto";

// A location where data is stored, including match data.
message Location {
  // The location in different potential backing stores.
  oneof location {
    BlobStorageLocation blob_storage_location = 1;
  }
}

// Represents a location in object storage (eg. GCS, S3, ...)
message BlobStorageLocation {
  // The name of the bucket in object storage.
  string bucket_name = 1;

  // The path to the location within the bucket.
  string path = 2;
}
